src/logic/currentSession/hooks/useCurrentSafeAddressSync.ts

```diff
@@ -5,12 +5,15 @@ import { useLocation } from 'react-router-dom'
 
 import { extractPrefixedSafeAddress } from 'src/routes/routes'
 import addCurrentSafeAddress from 'src/logic/currentSession/store/actions/addCurrentSafeAddress'
+import addCurrentShortName from '../store/actions/addCurrentShortName'
 
 export const useCurrentSafeAddressSync = (): void => {
   const dispatch = useDispatch<Dispatch>()
   const location = useLocation()
 
   useEffect(() => {
-    dispatch(addCurrentSafeAddress(extractPrefixedSafeAddress(location.pathname).safeAddress))
+    const { shortName, safeAddress } = extractPrefixedSafeAddress(location.pathname)
+    dispatch(addCurrentShortName(shortName))
+    dispatch(addCurrentSafeAddress(safeAddress))
```

Comment:
While testing I noticed that we update the current safe address more than we need to i.e. when switching networks due to redirects. Should we only update if its not empty?

Reply:
I've just added a comparison, only dispatching when there is a difference between the URL and current store value.